source/socks5-support/tests/README.rst

```diff
@@ -0,0 +1,71 @@
+====================
+SOCKS5 Support Tests
+====================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes how drivers should test support for SOCKS5 proxies.
+
+Testing Requirements
+====================
+
+For a single server version (at least 5.0), drivers MUST add two
+Evergreen tasks: one with a replica set with TLS enabled, and one
+with TLS disabled. The servers of the replica set may listen on any free ports.
+
+Each task MUST also start up two SOCKS5 proxies: one requiring authentication,
+and one that does not require authentication.
+
+SOCKS5 Proxy Configuration
+--------------------------
+
+Drivers MUST use the ``socks5srv.py`` script in ``drivers-evergreen-tools``
+to run the SOCKS5 proxy servers for Evergreen tasks. This script MUST
+be run after the backing replica set has already been started,
+and MUST be started with ``--map "localhost:12345 to <host>"` where
+``host` is the host:port identifier of an arbitrary member of the replica set.
+The SOCKS5 proxy server requiring authentication MUST be started with
+``--port 1080 --auth username:p4ssw0rd`, the one not requiring authentication
+with `--port 1081``.
+
+Prose Tests
+===========
+
+Drivers MUST test the following connection strings:
+
+.. list-table::
+   :header-rows: 1
+
+   * - SOCKS5 auth required
+     - SOCKS5 auth disabled
+   * - :code:`mongodb://localhost:12345/?proxyHost=localhost&proxyPort=1080&directConnection=true` (fails)
```

Comment:
Since we use `<replicaset>` just below, for consistency consider using <first seed list member> here instead of `localhost:12345`